### PR TITLE
Add Checkstyle via Gradle

### PR DESCRIPTION
**Pull request by NicMcPhee**
This adds the Checkstyle plugin via Gradle, and configures it so that it doesn't complain about some things that would be pretty annoying.

### CODE REVIEW

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  <!-- See https://checkstyle.org/config_sizes.html -->
+  <module name="FileLength"/>
+  <module name="LineLength">
+    <property name="fileExtensions" value="java"/>
+    <property name="max" value="120"/>
+  </module>
+
+  <!-- Checks for whitespace                               -->
+  <!-- See https://checkstyle.org/config_whitespace.html -->
+  <module name="FileTabCharacter"/>
```
Comment (floogulinc) [inline config/checkstyle/checkstyle.xml:82]: should we move this down to the other `Checks for whitespace` block?

**Comment by floogulinc** [comment]
So since this is now part of the Gradle `check` task I guess we don't need to add another GitHub Action for it..

File: config/checkstyle/checkstyle.xml (hunk 1 above)
Comment (NicMcPhee) [reply config/checkstyle/checkstyle.xml:82]: We could. I lifted this from the an example, so I don't know if there's any particular reason for this ordering.

**Comment by NicMcPhee** [comment]
I'm still not happy with how this is working, and I don't want to force it and then end up breaking things in the very first lab, so I'm going to step away for now and focus on other things.

We might want to bring this idea back in the Javalin lab for this semester; if someone wants to poke at that feel free. Otherwise maybe I'll return to it over the summer.

**Comment by helloworld12321** [comment]
@NicMcPhee To stress-test the checkstyle, would you like to try adding it to the iteration template first?

Then we might have a better idea of what edge cases can arise / what rules we might want to turn off.

**Comment by NicMcPhee** [comment]
> @NicMcPhee To stress-test the checkstyle, would you like to try adding it to the iteration template first?
> 
> Then we might have a better idea of what edge cases can arise / what rules we might want to turn off.
> 
> (For example, we may consider turning [`JavadocParagraph`](https://checkstyle.sourceforge.io/config_javadoc.html#JavadocParagraph) off, since we tend to write our Javadoc in plaintext, not HTML 🙂)

A good idea. I probably won't do that right now, but I'll add it to my todo list and try to get to it later in the week.

**Comment by NicMcPhee** [comment]
[We got this added to the iteration template](https://github.com/UMM-CSci-3601/3601-iteration-template/pull/602) and it seemed to have worked out OK. I'm going to copy the Checkstyle config file we put together in that repo over to this repo and then tidy up the Java code so that the checks pass.

**Comment by NicMcPhee** [comment]
It turns out that when I first set Checkstyle up in this PR, I used the Google standards, where I started with the Sun standards in the iteration template. 

I think the Google standards are "stricter", so switching to the Sun standards is probably a reasonable choice since students will probably find even those pretty annoying.

It turned out that the code passed the our modification to the Sun standards as is, which is nice. So I think we're ready for review and merge!

> (For example, we may consider turning [`JavadocParagraph`](https://checkstyle.sourceforge.io/config_javadoc.html#JavadocParagraph) off, since we tend to write our Javadoc in plaintext, not HTML 🙂)

All the Javadoc checks are commented out in this version, and I don't think _I_ did that (I think it came that way), so we're good on that.

**Review by helloworld12321 — APPROVED** [review]
